torch/csrc/utils/python_arg_parser.h

```diff
@@ -273,6 +273,9 @@ inline const THPLayout& PythonArgs::layout(int i) {
 }
 
 inline int64_t PythonArgs::toInt64(int i) {
+  if (!signature.params[i].optional) {
```

Comment:
I would prefer following the pattern used in `dtype(int i)` and `dtypeWithDefault(...)`. (Invert the control flow so that `toInt64WithDefault` calls `toInt64` instead.)